src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php

```diff
@@ -24,17 +25,22 @@ class SimpleAuthenticationProvider implements AuthenticationProviderInterface
     private $simpleAuthenticator;
     private $userProvider;
     private $providerKey;
+    private $userChecker;
 
-    public function __construct(SimpleAuthenticatorInterface $simpleAuthenticator, UserProviderInterface $userProvider, $providerKey)
+    public function __construct(SimpleAuthenticatorInterface $simpleAuthenticator, UserProviderInterface $userProvider, $providerKey, UserCheckerInterface $userChecker)
```

Comment:
must be default null for BC

Reply:
Should it use `Symfony\Component\Security\Core\User\UserChecker` as default if null is present?
This would leave the method as is (without if statements)?